generator/src/main/java/com/algorand/sdkutils/DoGenerate.java

```diff
@@ -0,0 +1,159 @@
+package com.algorand.sdkutils;
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import com.algorand.sdkutils.generators.OpenApiParser;
+import com.algorand.sdkutils.generators.Utils;
+import com.algorand.sdkutils.listeners.GoGenerator;
+import com.algorand.sdkutils.listeners.JavaGenerator;
+import com.algorand.sdkutils.listeners.Publisher;
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.Parameters;
+import com.fasterxml.jackson.databind.JsonNode;
+
+
+/**
+ *
+ * Usage examples:
+ * Go:
+ * -g go
+ * --algodspec /Users/shantkarakashian/go/src/github.com/algorand/go-algorand/daemon/algod/api/algod.oas2.json
+ * --indexerspec /Users/shantkarakashian/go/src/github.com/algorand/indexer/api/indexer.oas2.json
+ * -m application
+ * --filter
+ * 
+ *  Java:
+ *  -g java
+ *  --algodspec /Users/shantkarakashian/go/src/github.com/algorand/go-algorand/daemon/algod/api/algod.oas2.json
+ *  --indexerspec /Users/shantkarakashian/go/src/github.com/algorand/indexer/api/indexer.oas2.json
+ *  AccountStateDelta,Application,ApplicationLocalState,ApplicationLocalStates,ApplicationParams,ApplicationResponse,ApplicationStateSchema,ApplicationsResponse,CatchpointAbortResponse,CatchpointStartResponse,CompileResponse,DryrunRequest,DryrunResponse,DryrunSource,DryrunState,DryrunTxnResult,EvalDelta,EvalDeltaKeyValue,StateSchema,TealKeyValue,TealValue,TransactionApplication,GetApplicationByID,GetAssetByID,TealCompile,TealDryrun,LookupApplicationByID,SearchForApplications,LookupApplication
+ *  
+ */
+public class DoGenerate {
```

Comment:
We could be added as subcommands of `Main.java` as well, there's a build target so that you can run without the IDE:
```
~/generator$ mvn package
      ...
~/generator$  java -jar target/generator-1.0.0-jar-with-dependencies.jar -h
Usage: <main class> [options] [command] [command options]
  Options:
    -h, --help

    -s, --specfile
      Path to specfile used by generator.
  Commands:
    java      Generate the Java Client SDK.
      Usage: java [options]
        Options:
        * -n, --client-name
            Full path to the OpenAPI v2 spec file.
        * -cp, --common-package
            Package name to put at the top of generated client files.
        * -c, --common-path
            Path where generated client files will be placed.
          -h, --help

        * -mp, --model-package
            Package name to put at the top of generated model files.
        * -m, --model-path
            Path where generated model files will be placed.
        * -pp, --paths-package
            Package name to put at the top of generated paths files.
        * -p, --paths-path
            Path where generated paths files will be placed.
          -s, --specfile
            Path to specfile used by generator.
        * -t, --token-name
            Name of the token used for this application. i.e. 
            X-Algo-API-Token. 
          -tr, --token-required
            Whether or not a no-token version of the constructor should be 
            created. 
            Default: false

    responses      Generate response test file(s).
      Usage: responses [options]
        Options:
          -f, --filter
            Only generate response files for objects which contain the given 
            value as a substring.
          -h, --help

        * -o, --output-dir
            Location to place response objects.
        * -p, --prefix
            Prefix to resulting response files.
          -s, --specfile
            Path to specfile used by generator.
```

Reply:
Let's talk about this later. I need to understand some more to do it.